crates/rattler_conda_types/src/conda_lock/builder.rs

```diff
@@ -0,0 +1,209 @@
+use crate::conda_lock::content_hash::CalculateContentHashError;
+use crate::conda_lock::{
+    content_hash, Channel, CondaLock, GitMeta, LockMeta, LockedDependency, Manager, PackageHashes,
+    TimeMeta, VersionConstraint,
+};
+use crate::{MatchSpec, Platform};
+use std::collections::{HashMap, HashSet};
+use url::Url;
+
+/// Struct used to build a conda-lock file
+#[derive(Default)]
+struct LockFileBuilder {
+    /// Channels used to resolve dependencies
+    pub channels: Vec<Channel>,
+    /// The platforms this lock file supports
+    pub platforms: HashSet<Platform>,
+    /// Paths to source files, relative to the parent directory of the lockfile
+    pub sources: Option<Vec<String>>,
+    /// Metadata dealing with the time lockfile was created
+    pub time_metadata: Option<TimeMeta>,
+    /// Metadata dealing with the git repo the lockfile was created in and the user that created it
+    pub git_metadata: Option<GitMeta>,
+
+    /// Keep track of locked packages per platform
+    pub locked_packages: HashMap<Platform, LockedPackages>,
+
+    /// MatchSpecs input
+    /// This is only used to calculate the content_hash
+    /// for the lock file
+    pub input_specs: Vec<MatchSpec>,
+}
+
+impl LockFileBuilder {
+    pub fn new(
+        channels: impl IntoIterator<Item = Channel>,
```

Comment:
Since you can construct a `Channel` from `String` or `&str` maybe you could do:

```suggestion
        channels: impl IntoIterator<Item = impl Into<Channel>>,
```

then you can also simply pass a `String` directly, like: `["conda-forge"]`

Reply:
Done. I wanted that initially anyways :)